loony/node.nim

```diff
@@ -5,10 +5,27 @@ import loony/memalloc
 
 type
   Node* = object
-    slots* : array[0..N, Atomic[uint]]  # Pointers to object
+    slots* : array[N, Atomic[uint]]  # Pointers to object
     next*  : Atomic[NodePtr]            # NodePtr - successor node
     ctrl*  : ControlBlock               # Control block for mem recl
 
+when defined(loonyDebug):
+  var nodeCounter* {.global.}: Atomic[int]
+  nodeCounter.store(0)
+
+  proc echoDebugNodeCounter*(expected: int = 0) =
+    ## This will output the counter if it is above the expected value
+    let counter = nodeCounter.load()
+    if counter > 0:
+      warn "Node counter: " & $nodeCounter.load()
+  
+  template debugNodeCounter*(body: untyped) =
+    let initC = nodeCounter.load()
+    body
+    let newC = nodeCounter.load()
+    if (newC - initC) > 0:
+      warn "Finished block with node count of " & $(nodeCounter.load() - initC)
+
```

Comment:
if you add
```nim
else:
  template debugNodeCounter(body: untyped): untyped = discard
```
...or whatever, then you can use it without the gratuitous `when defined(loonyDebug):`, right?